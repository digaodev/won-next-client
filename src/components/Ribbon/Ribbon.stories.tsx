import { Story, Meta } from '@storybook/react/types-6-0'

import Ribbon, { RibbonProps } from './Ribbon'

export default {
  title: 'Ribbon',
  component: Ribbon,
  argTypes: {
    children: {
      type: 'string'
    }
  },
  args: {
    children: 'Best Seller'
  }
} as Meta

const Template: Story<RibbonProps> = (args) => <Ribbon {...args} />

export const Default = Template.bind({})
