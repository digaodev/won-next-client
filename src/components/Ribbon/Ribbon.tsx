import * as S from './Ribbon.styles'

export type RibbonColors = 'primary' | 'secondary'
export type RibbonSizes = 'normal' | 'small'

export type RibbonProps = {
  /**
   * Required children
   */
  children: React.ReactNode

  /**
   * Optional background color
   */
  color?: RibbonColors

  /**
   * Optional size
   */
  size?: RibbonSizes
}

const Ribbon = ({
  children,
  color = 'primary',
  size = 'normal'
}: RibbonProps) => (
  <S.Wrapper color={color} size={size}>
    {children}
  </S.Wrapper>
)

export default Ribbon
