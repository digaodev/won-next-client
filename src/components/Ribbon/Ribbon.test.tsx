import { screen } from '@testing-library/react'

import { renderWithTheme } from 'utils/tests/helpers'

import Ribbon from './Ribbon'

describe('<Ribbon />', () => {
  it('should render the Ribbon with a text', () => {
    const { container } = renderWithTheme(<Ribbon>Best Seller</Ribbon>)

    expect(screen.getByText(/Best Seller/i)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the Ribbon with the primary color as default', () => {
    renderWithTheme(<Ribbon>Best Seller</Ribbon>)

    expect(screen.getByText(/Best Seller/i)).toHaveStyle({
      backgroundColor: '#F231A5'
    })
  })

  it('should render the Ribbon with the secondary color', () => {
    renderWithTheme(<Ribbon color="secondary">Best Seller</Ribbon>)

    expect(screen.getByText(/Best Seller/i)).toHaveStyle({
      backgroundColor: '#3CD3C1'
    })
  })

  it('should render the Ribbon with the normal size as default', () => {
    renderWithTheme(<Ribbon>Best Seller</Ribbon>)

    expect(screen.getByText(/Best Seller/i)).toHaveStyle({
      height: '3.6rem',
      fontSize: '1.4rem'
    })
  })

  it('should render the Ribbon with the small size', () => {
    renderWithTheme(<Ribbon size="small">Best Seller</Ribbon>)

    expect(screen.getByText(/Best Seller/i)).toHaveStyle({
      height: '2.6rem',
      fontSize: '1.2rem'
    })
  })
})
