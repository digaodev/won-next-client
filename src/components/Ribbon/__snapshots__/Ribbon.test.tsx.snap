// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ribbon /> should render the Ribbon with a text 1`] = `
.c0 {
  position: absolute;
  top: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FAFAFA;
  font-weight: 600;
  background-color: #F231A5;
  right: -2rem;
  font-size: 1.4rem;
  height: 3.6rem;
  padding: 0 2.4rem;
}

.c0::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0rem;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

.c0::before {
  border-left-color: #b20b6f;
  border-top-color: #b20b6f;
}

.c0::before {
  top: 3.6rem;
  border-right-width: 2rem;
  border-top-width: 1rem;
}

<div
  class="c0"
  color="primary"
>
  Best Seller
</div>
`;
